File: src/main/java/com/dec/decisland/item/custom/HardBambooKatana.java
package com.dec.decisland.item.custom;

import com.dec.decisland.particles.ModParticles;
import net.minecraft.core.component.DataComponents;
import net.minecraft.nbt.CompoundTag;
import net.minecraft.server.level.ServerLevel;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.component.CustomData;

public class HardBambooKatana extends BambooKatana {
    public HardBambooKatana(Properties properties) {
        super(properties);
    }
    @Override
    protected int getMaxAttackCount(){
        return 6;
    }
    /**
     * 在视野前方生成粒子
     *
     * @param serverLevel
     * @param x
     * @param y
     * @param z
     */
    @Override
    public void useSpawnParticle(ServerLevel serverLevel, double x, double y, double z) {
        serverLevel.sendParticles(ModParticles.HARD_BAMBOO_KATANA_PARTICLE.get(), x, y, z, 1, 0, 0, 0, 0);
    }
    @Override
    public void attackSpawnParticle(ServerLevel serverLevel, double x, double y, double z) {
        serverLevel.sendParticles(ModParticles.HARD_BAMBOO_KATANA_PARTICLE.get(), x, y, z, 1, 0, 0, 0, 0);
    }
}

File: src/main/java/com/dec/decisland/item/custom/HardBambooKatana.kt
package com.dec.decisland.item.custom

import com.dec.decisland.DecIsland
import net.minecraft.resources.Identifier
import net.minecraft.server.level.ServerLevel

class HardBambooKatana(properties: Properties) : BambooKatana(properties) {
    override fun getMaxAttackCount(): Int = 6

    override fun useSpawnParticle(serverLevel: ServerLevel, x: Double, y: Double, z: Double) {
        sendBedrockKatanaEmitter(serverLevel, HARD_BAMBOO_KATANA_PARTICLE_ID, x, y, z)
    }

    override fun attackSpawnParticle(serverLevel: ServerLevel, x: Double, y: Double, z: Double) {
        sendBedrockKatanaEmitter(serverLevel, HARD_BAMBOO_KATANA_PARTICLE_ID, x, y, z)
    }

    companion object {
        private val HARD_BAMBOO_KATANA_PARTICLE_ID: Identifier =
            Identifier.fromNamespaceAndPath(DecIsland.MOD_ID, "hard_bamboo_katana_particle")
    }
}
